### PR TITLE
Add migration notes for log formatters.

### PR DESCRIPTION
**Pull request by othercorey**
Refs https://github.com/cakephp/cakephp/pull/15543

### CODE REVIEW

File: en/appendices/4-3-migration-guide.rst [hunk 1]
```diff
@@ -97,6 +107,18 @@ Http
 - The ``CspMiddleware`` now sets the ``cspScriptNonce`` and ``cspStyleNonce``
   request attributes which streamlines the adoption of strict
   content-security-policy rules.
+  
+Log
+---
```
Comment (markstory) [inline en/appendices/4-3-migration-guide.rst:112]: I can add a section to the logging docs to give a guide on how to use formatted and build new ones.

Comment (othercorey) [reply en/appendices/4-3-migration-guide.rst:112]: Should I reduce the migration notes here? I don't know if it's overly verbose.

File: en/core-libraries/logging.rst [hunk 2]
```diff
@@ -356,6 +206,174 @@ Note that using this form will limit the ability to pass more data as context::
...
 
+.. _file-log:
+
+Logging to Files
+================
+
+As its name implies ``FileLog`` writes log messages to files. The level of log
+message being written determines the name of the file the message is stored in.
+If a level is not supplied, :php:const:`LOG_ERR` is used which writes to the
+error log. The default log location is ``logs/$level.log``::
```
Comment (othercorey) [inline en/core-libraries/logging.rst]:
```suggestion
error log. The default log location is **logs/$level.log**::
```

paths are typically bold, right?

File: en/core-libraries/logging.rst [hunk 3]
```diff
@@ -356,6 +206,174 @@ Note that using this form will limit the ability to pass more data as context::
...
+properties are passed to the log engine's constructor as an array. ::
+
+    namespace App\Log\Engine;
+    use Cake\Log\Engine\BaseLog;
+
+    class DatabaseLog extends BaseLog
+    {
+        public function __construct($options = [])
+        {
+            parent::__construct($options);
```
Comment (othercorey) [inline en/core-libraries/logging.rst]:
```suggestion
        public function __construct(array $config= [])
        {
            parent::__construct($config);
```

File: en/core-libraries/logging.rst [hunk 4]
```diff
@@ -356,6 +206,174 @@ Note that using this form will limit the ability to pass more data as context::
...
+            // ...
+        }
+
+        public function log($level, $message, array $context = [])
```
Comment (othercorey) [inline en/core-libraries/logging.rst]:
```suggestion
        public function log($level, string $message, array $context = [])
```

File: en/core-libraries/logging.rst [hunk 5]
```diff
@@ -356,6 +206,174 @@ Note that using this form will limit the ability to pass more data as context::
...
+When configuring a log engine the ``className`` parameter is used to
+locate and load the log handler. All of the other configuration
+properties are passed to the log engine's constructor as an array. ::
+
+    namespace App\Log\Engine;
+    use Cake\Log\Engine\BaseLog;
+
+    class DatabaseLog extends BaseLog
+    {
+        public function __construct(array $config= [])
```
Comment (garas) [inline en/core-libraries/logging.rst]:
```suggestion
        public function __construct(array $config = [])
```

**Pull request merged**